Api/ExceptionHandler/GlobalExceptionHandling.cs

```diff
@@ -0,0 +1,57 @@
+﻿namespace Api.ExceptionHandler
+{
+    using Application.Exceptions;
+    using System.Net;
+    using System.Text.Json;
+
+    public class GlobalExceptionHandlingMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ILogger<GlobalExceptionHandlingMiddleware> _logger;
+
+        public GlobalExceptionHandlingMiddleware(RequestDelegate next, ILogger<GlobalExceptionHandlingMiddleware> logger)
+        {
+            _next = next;
+            _logger = logger;
+        }
+
+        public async Task InvokeAsync(HttpContext httpContext)
+        {
+            try
+            {
+                await _next(httpContext);
+            }
+            catch (Exception ex)
+            {
+                await HandleExceptionAsync(httpContext, ex);
+            }
+        }
+
+        private async Task HandleExceptionAsync(HttpContext context, Exception exception)
+        {
+            context.Response.ContentType = "application/json";
+            var response = context.Response;
+
+            var errorResponse = new ErrorResponse
+            {
+                Success = false
+            };
+            switch (exception)
+            {
+                case ClientException ex:
+                    response.StatusCode = (int)HttpStatusCode.BadRequest;
+                    errorResponse.Message = ex.Message;
+                    break;
+
+
+                default:
+                    response.StatusCode = (int)HttpStatusCode.InternalServerError;
+                    errorResponse.Message = "Internal server error!";
+                    break;
+            }
+            _logger.LogError(exception, exception.Message);
```

Comment:
I think it's better if we pass: "Unhandled Application Exception" as message so it's clear that the exception was logged by this middleware.